problem-solver/cxx/inferenceModule/searcher/SolutionTreeSearcher.cpp

```diff
@@ -0,0 +1,65 @@
+#include "SolutionTreeSearcher.hpp"
+#include "keynodes/InferenceKeynodes.hpp"
+
+namespace inference
+{
+SolutionTreeSearcher::SolutionTreeSearcher(ScMemoryContext * context)
+    : context(context)
+{
+}
+
+bool SolutionTreeSearcher::checkIfSolutionNodeExists(const ScAddr & rule, ScTemplateParams const & templateParams, std::set<std::string> const & varNames)
```

Comment:
const after type everywhere
```suggestion
bool SolutionTreeSearcher::checkIfSolutionNodeExists(ScAddr const & rule, ScTemplateParams const & templateParams, std::set<std::string> const & varNames)
```